File: C++/Medium/RuntimeError/count-the-number-of-fair-pairs_8.cpp
class Solution {
public:
    long long countFairPairs(vector<int>& nums, int lower, int upper) {
        
        long long result = 0;
        vector<int> count(upper + 1, 0);
        for (int i = 0; i < nums.size(); i++) {
            int left = max(0, lower - nums[i]);
            int right = min(upper, upper - nums[i]);
            for (int j = left; j <= right; j++) {
                result += count[j];
            }
            count[nums[i]]++;
        }
        return result;
    }
};

File: Java/Medium/WrongAnswer/count-the-number-of-fair-pairs_8.java
class Solution {
    public long countFairPairs(int[] nums, int lower, int upper) {
        
        int n = nums.length;
        int[] count = new int[200001];
        int[] sum = new int[n];
        sum[0] = nums[0];
        for (int i = 1; i < n; ++i) {
            sum[i] = sum[i - 1] + nums[i];
        }
        long ans = 0;
        for (int i = 0; i < n; ++i) {
            int l = lower - sum[i], r = upper - sum[i];
            if (l <= 0 && r >= 0) {
                ans += count[100000];
            }
            if (l > 0) {
                ans += count[100000 + l];
            }
            if (r < 0) {
                ans += count[100000 + r + 1];
            }
            ++count[sum[i] + 100000];
        }
        return ans;
    }
}